feat(plugin-cassandra): Upgrade to Cassandra Java Driver 4.x (#27029)
prestodb/presto: presto-docs/src/main/sphinx/connector/cassandra.rst

Reviewer: ```suggestion
.. _Getting Started: https://cassandra.apache.org/doc/latest/cassandra/getting-started/index.html
```
Noticed the Getting Started link was broken, fixed it.

Author: good catch and thank you for reviewing!

                                                   driver will use these contact points to discover cluster topology.
                                                   At least one Cassandra host is required.

``cassandra.cloud.secure-connect-bundle``          Path to the secure connect bundle for connecting to managed services
                                                   cloud. This is required when connecting to Astra DB in the cloud.
                                                   ``cassandra.contact-points`` should be avoided when this property is set.

>                                                    More information about secure connect bundle can be found in the

Reviewer: ```suggestion
                                                   More information about secure connect bundle can be found in 
```

>                                                    `Use Secure Connect Bundles`_. This property is optional.

Reviewer: ```suggestion
                                                   `Download and use a Secure Connect Bundle (SCB) with an Astra DB Serverless database`_. This property is optional.
```
I wrote, but didn't post, this comment - which pairs with the recommended change in line 104.

        If authorization is enabled, ``cassandra.username`` must have enough permissions to perform ``SELECT`` queries on
        the ``system.size_estimates`` table.

.. _Cassandra consistency: https://docs.datastax.com/en/cassandra-oss/3.x/cassandra/dml/dmlConfigConsistency.html

> .. _Use Secure Connect Bundles: https://docs.datastax.com/en/astra-db-serverless/databases/secure-connect-bundle.html

Reviewer: ```suggestion
.. _Download and use a Secure Connect Bundle (SCB) with an Astra DB Serverless database: https://docs.datastax.com/en/astra-db-serverless/databases/secure-connect-bundle.html
```

  query with a partition key as a filter.
* ``IN`` list filters are only allowed on index (that is, partition key or clustering key) columns.
* Range (``<`` or ``>`` and ``BETWEEN``) filters can be applied only to the partition keys.

Migration from Driver 3.x
-------------------------

If you are upgrading from a previous version of Presto that used Cassandra Java Driver 3.x, please note the following changes:

Removed Configuration Properties

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

The following configuration properties are **no longer supported** in driver 4.x and must be removed from your configuration:

* ``cassandra.load-policy.use-white-list`` - WhiteListPolicy does not exist in driver 4.x
* ``cassandra.load-policy.white-list.addresses`` - WhiteListPolicy does not exist in driver 4.x

**If you were using whitelist policy**, you have these alternatives:

1. **Network-level filtering**: Use firewall rules or network policies to restrict which Cassandra nodes are accessible.

2. **Contact points selection**: Specify only the desired nodes in ``cassandra.contact-points``. The driver will discover the cluster topology from these nodes.

3. **Custom load balancing**: Implement a custom load balancing policy if you need advanced node filtering logic.

**Action Required**: Remove these properties from your ``cassandra.properties`` configuration file:

.. code-block:: properties

>     # REMOVE THESE LINES if present:

Reviewer: Does this line need to be part of the code block? Does it need to be here at all?

Author: yes, it needs to be here recommending the users to remove these properties, if present on their existing properties file/codebase

Reviewer: I ask because this line seems redundant with the instruction on line 302 
```
**Action Required**: Remove these properties from your ``cassandra.properties`` configuration file:
```

Author: that 302 starts the call to attention, followed by the code blocks which specify the exact properties to be removed. Sorry, am I missing something here?

Reviewer: No, I think I was missing something. :) Thanks!

    # cassandra.load-policy.use-white-list=true
    # cassandra.load-policy.white-list.addresses=node1,node2,node3

Protocol Version Changes
^^^^^^^^^^^^^^^^^^^^^^^^

* Protocol V2 is no longer supported. Minimum supported version is V3.

> * The driver now automatically negotiates the highest common protocol version, so manual configuration is rarely needed.

Reviewer: ```suggestion
* The driver automatically negotiates the highest common protocol version, so manual configuration is rarely needed.
```

> * Protocol V5 is now supported for Cassandra 4.0+.

Reviewer: ```suggestion
* Protocol V5 is supported for Cassandra 4.0+.
```
